### ISSUE TITLE
Usage of legacy context API

### ISSUE THREAD

**Issue by thetric** [reactstrap/reactstrap#1340]
- components: `TabPane`, `Dropdown(,Item,Menu,Toggle)`, `Manager`, `PopperTargetHelper`, `CarousellItem` (see https://github.com/reactstrap/reactstrap/search?q=contextTypes&unscoped_q=contextTypes)
- reactstrap version `6.5.0`
- import method: `es`
- react version `16.7.0`
- bootstrap version `4.2.1`

### What is happening?
When using `<React.StrictMode>` Reacts advises against using the legacy context API:

```
Warning: Legacy context API has been detected within a strict-mode tree: 
    in StrictMode (at App.tsx:27)
    in App (at src/index.tsx:7)

Please update the following components: Dropdown, DropdownItem, DropdownMenu, DropdownToggle, Manager

Learn more about this warning here:
https://fb.me/react-strict-mode-warnings
```

> _From https://reactjs.org/docs/context.html#legacy-api:_
>
> React previously shipped with an experimental context API. The old API will be supported in all 16.x releases, but applications using it should migrate to the new version. The legacy API will be removed in a future major React version. [...]

### What should be happening?
Reactstrap should be using the new Context API.

### Steps to reproduce issue
Wrap an example app using some of the components mentions at the beginning in `React.StrictMode` and start it in development mode.

### Code

https://stackblitz.com/edit/reactstrap-saxswj?file=Example.js

**Comment by thetric** [comment]
The `Modal` component uses deprecated lifecycle hooks:

```
componentWillReceiveProps: Please update the following components to use static getDerivedStateFromProps instead: Modal

componentWillUpdate: Please update the following components to use componentDidUpdate instead: Modal
```

**Comment by gergely-nagy** [comment]
I'll submit a PR for it!

**Comment by iegik** [comment]
Duplicate of https://github.com/reactjs/react-transition-group/issues/429

**Comment by armenzg** [comment]
@gergely-nagy any progress?

**Comment by jonathan-richer** [comment]
Same thing on `Collapse`:
![image](https://user-images.githubusercontent.com/36356746/83686057-9378ff00-a5b7-11ea-92f9-ef3691d2c1db.png)

**Comment by sumanthratna** [comment]
I got this with `Tooltip`, if that helps.

**Comment by MA-Ahmad** [comment]
I got this warning with Carousel.

**Comment by iamandrewluca** [comment]
Anyone interested in fixing this?
Here Is a starting point https://github.com/reactstrap/reactstrap/pull/1866
One test is failing. I can't alocate more time.

**Comment by kindyALkindy** [comment]
I got this warning with Tooltip

**Comment by rmlevangelio** [comment]
Also happening in Fade component

**Comment by eka-dyah** [comment]
I got this warning with Collapse navbar.

**Comment by thchong-code** [comment]
Got this warning in 
`in Transition (created by Collapse)
    in Collapse (at AppNavbar.js:24)
    in div (created by Container)
    in Container (at AppNavbar.js:21)
    in nav (created by Navbar)
    in Navbar (at AppNavbar.js:20)
    in div (at AppNavbar.js:19)
    in AppNavbar (at App.js:10)
    in div (at App.js:9)
    in App (at src/index.js:8)
    in StrictMode (at src/index.js:7)`

**Comment by iamandrewluca** [comment]
## We know about this issue.

### Legacy context is used only in carousel right now, no other places.
### All other warnings come from `react-transition-group` if you enable `StrictMode`, because it uses deprecated `findDOMNode` API

### There is no more reasons to post about it.

**Comment by vishaloswal** [comment]
> Same thing on `Collapse`:
> ![image](https://user-images.githubusercontent.com/36356746/83686057-9378ff00-a5b7-11ea-92f9-ef3691d2c1db.png)

Even I get similar warning, is there a work around or fix for this?

**Comment by SumitJadiya** [comment]
Workaround:
Remove strictMode from index.js

**Comment by xreider** [comment]
It is not resolved yet

**Comment by iamandrewluca** [comment]
@xreider do you want to help fixing it?

**Comment by manhddse03539** [comment]
In index.js change `React.StrictMode` to`React.Fragment` @@

**Comment by chauger** [comment]
ALert is causing this issue for me

**Comment by adamf1631** [comment]
switch to <React.Fragment> in index.js

**Comment by iamandrewluca** [comment]
`React.Fragment` is not an alternative for `React.StrictMode` 🙂

**Comment by d3ce1t** [comment]
@iamandrewluca CMIIW the problem comes from react-transition-groups (as you said). Reactstrap is using that library at version 2.9.0. But last version is 4.4.1. Would it still happen in a more recent version? What's blocking reactstrap from updating?

**Comment by thetric** [comment]
My guess is that the author wants to support React < 16.3 which has no the new Context API. But `react-transition-groups@3` [requires the new Context API](https://github.com/reactjs/react-transition-group/blob/master/CHANGELOG.md#300-2019-04-15)

**Comment by iamandrewluca** [comment]
We don't have time to allocate for this. And because this will be a breaking change, we thought of making multiple breaking changes at a time. But for that we have to allocate a lot of time, which we don't have.

**Comment by adamf1631** [comment]
Yes, I found this warning as well, if you boys have any suggestions please step up and suggest. Thanks!

**Comment by mazhar1987** [comment]
Same problem I'm facing when using Collapse from reactstrap. Is there any solution? Thanks

<img width="1431" alt="Screenshot 2021-02-09 at 9 16 35 PM" src="https://user-images.githubusercontent.com/3949434/107384558-47a23080-6b1c-11eb-9c23-0d13a17b1e9e.png">

**Comment by CleverOscar** [comment]
Has this issue really been on for more than two years? I keep getting an error when using the `Carousel` component

**Comment by dogukankutluay** [comment]
> İndex.js değişikliklerinde `React.StrictMode`için `React.Fragment`@@

thankx :)

**Comment by musasahinkundakci** [comment]
`React.Fragment` if you are beginner you can use it temporily :)

**Comment by fsogutcen** [comment]
Deleting "<React.Fragment>" in the index.js file may also be effective.
I chose to use it this way to solve it; "ReactDOM.render (<App />,document.getElementById('root '));"

But since I'm newbie to React, I'm not sure how the right step is

**Comment by fsogutcen** [comment]
Deleting "<React.Fragment>" in the index.js file may also be effective.
I chose to use it this way to solve it;  " ReactDOM.render(<App />,document.getElementById('root')); "

But since I'm newbie to React, I'm not sure how the right step is

**Comment by dogukankutluay** [comment]
> İndex.js dosyasında "<React.Fragment>" i silmek de etkili olabilir.
> Bunu çözmek için bu şekilde kullanmayı seçtim; "ReactDOM.render (, document.getElementById ('root'));"
> 
> Ama React için acemi olduğum için doğru adımın nasıl olduğundan emin değilim

selamlar , direkt olarak bütün dosyalar için fragment verilmesi en mantıklısı. İhtimale dayalı bir kod aslında. Araştırdım biraz

**Comment by sbaechler** [comment]
@iamandrewluca Can the removal of the legacy APIs be part of the 9.0 release?

**Comment by Allakazan** [comment]
This issue still are going on ? Isn't unsafe and dangerous to disable StrictMode ?

**Comment by bestguy** [comment]
@sbaechler , @Allakazan yes we have an opportunity now with the upcoming 9.x release.  
However FYI there are no full time maintainers so any PRs for this off [bootstrap5 branch](https://github.com/reactstrap/reactstrap/tree/bootstrap5) would be appreciated.
(A few of us are helping out to get Reactstrap 9.x Bootstrap 5.x stable and released.)

**Comment by bestguy** [comment]
I've updated rtg to v4 in linked PR, but need to test if addresses everything in this issue.  I don't see direct usages of old context API, but catching up here.

**Comment by bestguy** [comment]
Also updated Carousel context issue in #2268 
We'll do another pre release this week for you all to try out.
